### PR TITLE
feat(dotnet): publish constructs to NuGet

### PR DESCRIPTION
**Pull request by dineshSajwan**
Fixes #
1. Create a .Net package for generative ai cdk constructs and push it to NuGet.
---
By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the project license.

### CODE REVIEW

File: .projenrc.ts [hunk 1]
```diff
@@ -14,6 +14,7 @@ import { ProjenStruct, Struct } from '@mrgrain/jsii-struct-builder';
 import { JsonPatch, awscdk } from 'projen';
 import { DependabotScheduleInterval, VersioningStrategy } from 'projen/lib/github';
 import { NpmAccess } from 'projen/lib/javascript';
+import { ReleaseTrigger } from 'projen/lib/release';
```
Comment (scottschreckengaust) [inline .projenrc.ts:17]:
```suggestion
```

File: .projenrc.ts [hunk 2]
```diff
@@ -74,12 +75,20 @@ const project = new awscdk.AwsCdkConstructLibrary({
   npmTokenSecret: 'NPM_TOKEN',
   npmAccess: NpmAccess.PUBLIC,
 
+  // Manual release, remove once NuGet Account is activated
+  releaseTrigger: ReleaseTrigger.manual(),
+
```
Comment (scottschreckengaust) [inline .projenrc.ts:80]:
```suggestion
```

File: .projenrc.ts [hunk 3]
```diff
@@ -74,12 +75,20 @@ const project = new awscdk.AwsCdkConstructLibrary({
...
   publishToPypi: {
     distName: PUBLICATION_NAMESPACE+'.'+PROJECT_NAME,
     module: (PUBLICATION_NAMESPACE.replace(/-/g, '_'))+'.'+(PROJECT_NAME.replace(/-/g, '_')), // PEP 8, convert hypens
     // twineRegistryUrl: '${{ secrets.TWINE_REGISTRY_URL }}',
   },
 
+  publishToNuget: {
+    dotNetNamespace: PUBLICATION_NAMESPACE+'.'+PROJECT_NAME.replace(/-/g, '_'),
+    packageId: PUBLICATION_NAMESPACE+'.'+PROJECT_NAME.replace(/-/g, '_'),
```
Comment (scottschreckengaust) [inline .projenrc.ts:89]: We may want to CamelCase this...let me test...probably add a function somewhere to define this CamelCasing

```suggestion
    dotNetNamespace: PUBLICATION_NAMESPACE.replace(/-/g, ' ').replace(/_/g, ' ').replace(/(?:^\w|[A-Z]|\b\w|\s+)/g, function(match, index) {    if (+match === 0) return ""; return match.toUpperCase();  })+'.'+PROJECT_NAME.replace(/-/g, '_'),
    packageId: PUBLICATION_NAMESPACE.replace(/-/g, ' ').replace(/_/g, ' ').replace(/(?:^\w|[A-Z]|\b\w|\s+)/g, function(match, index) {    if (+match === 0) return ""; return match.toUpperCase();  })+'.'+PROJECT_NAME.replace(/-/g, ' ').replace(/_/g, ' ').replace(/(?:^\w|[A-Z]|\b\w|\s+)/g, function(match, index) {    if (+match === 0) return ""; return match.toUpperCase();  }),
```

Comment (scottschreckengaust) [reply .projenrc.ts:89]: Testing the package as-is first to see what it yields

**Review by scottschreckengaust — DISMISSED** [review]

File: .github/workflows/build.yml [hunk 4]
```diff
@@ -141,3 +141,31 @@ jobs:
...
+  package-dotnet:
+    needs: build
+    runs-on: ubuntu-latest
+    permissions: {}
+    if: "! needs.build.outputs.self_mutation_happened"
+    steps:
+      - uses: actions/setup-node@60edb5dd545a775178f52524783378180af0d1f8
+        with:
+          node-version: 20.x
+      - uses: actions/setup-dotnet@v3
```
Comment (krokoko) [inline .github/workflows/build.yml:153]: upgrade this to commit hash

Comment (krokoko) [reply .github/workflows/build.yml]: done in [4cc87aa](https://github.com/awslabs/generative-ai-cdk-constructs/pull/529/commits/4cc87aac1d2cb5d984ff82ee6259b592b3e56d5e) by @scottschreckengaust

**Review by krokoko — APPROVED** [review]

**Review by scottschreckengaust — APPROVED** [review]

**Pull request merged**